mesonbuild/minstall.py

```diff
@@ -431,10 +431,12 @@ def do_copyfile(self, from_file: str, to_file: str,
         append_to_log(self.lf, to_file)
         return True
 
-    def do_symlink(self, target: str, link: str, full_dst_dir: str, allow_missing: bool) -> bool:
+    def do_symlink(self, target: str, link: str, destdir: str, full_dst_dir: str, allow_missing: bool) -> bool:
         abs_target = target
         if not os.path.isabs(target):
             abs_target = os.path.join(full_dst_dir, target)
+        elif not os.path.exists(abs_target) and not allow_missing:
+            abs_target = os.path.join(destdir, abs_target[len(Path(abs_target).anchor):])
```

Comment:
This looks unfortunate(ly complex) and should consider using destdir_join instead.

Reply:
Ok, done.
But also rewritten `destdir_join` to use `pathlib` as it was suggested long time ago